cmd/peapod-to-fstree/main.go

```diff
@@ -0,0 +1,330 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-node/config"
+	engineconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine"
+	shardconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard"
+	fstreeconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard/blobstor/fstree"
+	objectcore "github.com/nspcc-dev/neofs-node/pkg/core/object"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/common"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/compression"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/fstree"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/peapod"
+	meta "github.com/nspcc-dev/neofs-node/pkg/local_object_storage/metabase"
+	"go.etcd.io/bbolt"
+	"gopkg.in/yaml.v3"
+)
+
+type epochState struct{}
+
+func (s epochState) CurrentEpoch() uint64 {
+	return 0
+}
+
+func main() {
+	nodeCfgPath := flag.String("config", "", "Path to storage node's YAML configuration file")
+
+	flag.Parse()
+
+	if *nodeCfgPath == "" {
+		log.Fatal("missing storage node config flag")
+	}
+
+	srcPath := *nodeCfgPath
+	ss := strings.Split(srcPath, ".")
```

Comment:
i would recomment smth like
```go
extension := filepath.Ext(srcPath)
name := srcPath[0:len(srcPath)-len(extension)]
```

dots are not prohibited in file names

Reply:
In general, you can just do `+ ".migrated"` and that's it. `srcPath` can be anything, with many dots, without dots, whatever.